mini-projects/00-jq-rs/src/processor.rs

```diff
@@ -0,0 +1,253 @@
+                    Some(needle) => output.push((Commands::Delete, String::from(&needle[1]))),
+                    None => {
+                        //check if we have a delete with an index
+                        match delete_index_regex.captures(string) {
+                            Some(needle) => {
+                                output.push((Commands::Delete, String::from(&needle[1])))
+                            }
+                            None => {
+                                //Check for an array slice
+                                match array_slice_regex.captures(string) {
+                                    Some(needle) => output.push((
+                                        Commands::Filter(FilterType::ArraySlice),
+                                        String::from(&needle[1]),
+                                    )),
+                                    None => {
+                                        //check if it is an array index
+                                        match array_index_regex.captures(string) {
+                                            Some(needle) => output.push((
+                                                Commands::Filter(FilterType::ArrayIndex),
+                                                String::from(&needle[1]),
+                                            )),
+                                            None => {
+                                                //finally we can check if it is an object index, if it isn't this than the input was invalid so the code will panic
+                                                let needle = object_regex.captures(string).unwrap();
+                                                output.push((
+                                                    Commands::Filter(FilterType::ObjectIdentity),
+                                                    String::from(&needle[1]),
+                                                ))
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    output
+}
+//takes in a list of commands and executes them, if there is an invalid command then this will trigger error handling in the filter code
+fn process(input: String, initial: Value) -> Result {
+    let list = filterneedle(input);
+    let mut input = Result::Value(initial);
+    for command in list {
+        //if the input for the next stage is anything other than a value we should panic and tell the user that they gave an invalid filter
+        match input {
+            Result::Value(ref val) => input = execute(command.0, command.1, val.clone()),
+            _ => panic!("Invalid input filter!"),
+        };
+    }
+    input
+}
+pub fn call_printer(
```

Comment:
super nit: this function seems slightly misnamed - I would think `call_printer` would only print, not also run the filters